### PR TITLE
propagate to MDC even if no Span is present, handle key key types gracefully

### PR DESCRIPTION
**Pull request by ivantopo**
These changes are based on the suggestion here: https://gitter.im/kamon-io/Kamon?at=5c34a77dbd592b2e6e6fbd46. The core of it is:
  - Ensure that context keys can be added to the MDC regardless of whether there is a Span in the context or not.
  - Allow any key to be put into the MDC, not only Option[String] as it is now.

### CODE REVIEW

File: src/main/scala/kamon/logback/instrumentation/AsyncAppenderInstrumentation.scala [hunk 1]
```diff
@@ -77,25 +77,30 @@ class AsyncAppenderInstrumentation {
...
+      if(span.context().traceID != IdentityProvider.NoIdentifier) {
+        MDC.put(AsyncAppenderInstrumentation.mdcTraceKey, span.context().traceID.string)
+        MDC.put(AsyncAppenderInstrumentation.mdcSpanKey, span.context().spanID.string)
+      }
 
       AsyncAppenderInstrumentation.mdcLocalKeys.foreach { localKey =>
-        currentContext.get(localKey).foreach { localKeyValue =>
-          MDC.put(localKey.name, localKeyValue)
+        val localKeyValue = currentContext.get(localKey) match {
+          case Some(value: String)  => value
```
Comment (chotai) [inline src/main/scala/kamon/logback/instrumentation/AsyncAppenderInstrumentation.scala]: Would it be worth unwrapping all `Option[_]`?
eg.
```
case Some(value)  => value.toString
```

**Review by chotai — APPROVED** [review]
Tested this with our service and is working as expected. Thank you very much for the PR 👍

**Comment by dpsoft** [comment]
@ivantopo @chotai LGTM!

File: src/main/scala/kamon/logback/instrumentation/AsyncAppenderInstrumentation.scala (hunk 1 above)
Comment (ivantopo) [reply src/main/scala/kamon/logback/instrumentation/AsyncAppenderInstrumentation.scala]: Indeed, I'll apply that, merge and publish. Thanks for the feedback!

**Pull request merged**